File: modules/animated-text/ios/AnimatedTextModule.swift
import ExpoModulesCore
import UIKit

public class AnimatedTextModule: Module {
  public func definition() -> ModuleDefinition {
    Name("AnimatedText")

    View(AnimatedTextView.self) {
      Prop("value") { (view: AnimatedTextView, prop: Double) in
        view.updateValue(value: prop)
      }
      Prop("color") { (view: AnimatedTextView, color: UIColor) in
        view.updateColor(color)
      }
    }
  }
}

File: modules/animated-text/android/src/main/java/expo/modules/animatedtext/AnimatedTextModule.kt
package expo.modules.animatedtext

import android.graphics.Color
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class AnimatedTextModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("AnimatedText")

    View(AnimatedTextView::class) {
      Prop("value") { view: AnimatedTextView, value: Double ->
        view.updateValue(value)
      }
      Prop("color") { view: AnimatedTextView, color: Color ->
        view.updateColor(color.toArgb())
      }
    }
  }
}
